packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/diagrams/DiagramFilterControllerTests.java

```diff
@@ -0,0 +1,203 @@
+                "ExpandCollapseDiagram"
+        );
+        return this.givenCreatedDiagramSubscription.createAndSubscribe(input);
+    }
+
+    @Test
+    @DisplayName("Given a diagram and a diagram filter view, when a tool collapsing nodes is invoked, then the diagram filter view is updated")
+    @Sql(scripts = {"/scripts/papaya.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {"/scripts/cleanup.sql"}, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, config = @SqlConfig(transactionMode = SqlConfig.TransactionMode.ISOLATED))
+    public void givenDiagramAndDiagramFilterWhenToolCollapsingNodeIsInvokedThenDiagramFilterViewIsUpdated() {
+        var diagramFlux = this.givenSubscriptionToExpandedCollapseDiagram();
+
+        AtomicReference<Diagram> diagramReference = new AtomicReference<>();
+        AtomicReference<String> expandedNodeId = new AtomicReference<>();
+
+        Consumer<DiagramRefreshedEventPayload> initialDiagramContentConsumer = payload -> Optional.of(payload)
+                .map(DiagramRefreshedEventPayload::diagram)
+                .ifPresentOrElse(diagram -> {
+                    diagramReference.set(diagram);
+                    diagram.getNodes().stream()
+                        .filter(node -> node.getCollapsingState().equals(CollapsingState.EXPANDED))
+                        .map(Node::getId)
+                        .findFirst()
+                        .ifPresent(expandedNodeId::set);
+                }, () -> fail("Missing diagram"));
+
+        StepVerifier.create(diagramFlux)
+            .consumeNextWith(initialDiagramContentConsumer)
+            .thenCancel()
+            .verify(Duration.ofSeconds(10));
+
+        var propertiesInput = new PropertiesEventInput(UUID.randomUUID(), PapayaIdentifiers.PAPAYA_PROJECT.toString(), List.of(diagramReference.get().getId()));
+        var diagramFilterFlux = this.diagramFilterEventSubscriptionRunner.run(propertiesInput);
+
+        Predicate<Object> formContentMatcher = object -> Optional.of(object)
+                .filter(DataFetcherResult.class::isInstance)
+                .map(DataFetcherResult.class::cast)
+                .map(DataFetcherResult::getData)
+                .filter(FormRefreshedEventPayload.class::isInstance)
+                .map(FormRefreshedEventPayload.class::cast)
+                .map(FormRefreshedEventPayload::form)
+                .filter(form -> {
+                    var diagramNavigator = new DiagramNavigator(diagramReference.get());
+                    var group = new FormNavigator(form)
+                        .page("ExpandCollapseDiagram")
+                        .group("Filter elements")
+                        .getGroup();
+                    var optTreeWidget = group.getWidgets().stream()
+                        .filter(TreeWidget.class::isInstance)
+                        .map(TreeWidget.class::cast)
+                        .findFirst();
```

Comment:
=you can continue using the form navigator to retrieve the tree widget and ensure that it is present

Reply:
I should know that 🙂
Done